### PR TITLE
Adds hint for date/time task fields

### LINKED ISSUE #2706: [Date and time tasks] UX Bug: UI hint missing on date and time tasks
The date and time fields currently appear identical to text fields, as opposed to the desired UX:

image

Note that date should be in the device's preferred date format.

@vittorino @amegantz for designs for time tasks. @kenstershiro FYI

### PR DESCRIPTION
**Pull request by anandwana001**
Fixes #2706 

@... PTAL?

Screenshot 2024-10-04 at 1 29 54 PM
Screenshot 2024-10-04 at 1 29 47 PM
Screenshot 2024-10-04 at 1 29 41 PM
Screenshot 2024-10-04 at 1 29 33 PM

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt [hunk 1]
```diff
@@ -54,6 +57,23 @@ class DateTaskFragment : AbstractTaskFragment<DateTaskViewModel>() {
...
+        val hint =
+          pattern
+            .replace("yyyy", "YYYY") // Full year
+            .replace("yy", "YY") // Short year
+            .replace("MMMM", "MMMM") // Full month name
+            .replace("MMM", "MMM") // Abbreviated month name
+            .replace("MM", "MM") // Month (two-digit numeric)
+            .replace("M", "MM") // Single-digit month to two-digit
+            .replace("dd", "DD") // Two-digit day
+            .replace("d", "DD") // Single-digit day to two-digit
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt:74]: Can this be replaced with `toUpper()`?

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/time/TimeTaskFragment.kt [hunk 2]
```diff
@@ -54,6 +57,25 @@ class TimeTaskFragment : AbstractTaskFragment<TimeTaskViewModel>() {
...
+        val hint =
+          if (timeFormat is SimpleDateFormat) {
+            timeFormat
+              .toPattern()
+              .replace("hh", "HH") // 12-hour format to 2-digit hour (12-hour clock)
+              .replace("h", "HH") // 12-hour format to 2-digit hour (12-hour clock)
+              .replace("HH", "HH") // 24-hour format
+              .replace("mm", "MM") // Minutes (two-digit)
+              .replace("ss", "SS") // Seconds (two-digit)
+              .replace("a", "AM/PM") // AM/PM indication with a space
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/time/TimeTaskFragment.kt:73]: Same as previous comment

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt (hunk 1 above)
Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt:74]: done

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/time/TimeTaskFragment.kt (hunk 2 above)
Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/time/TimeTaskFragment.kt]: done

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt [hunk 3]
```diff
@@ -54,6 +57,18 @@ class DateTaskFragment : AbstractTaskFragment<DateTaskViewModel>() {
...
         .asLiveData()
+
+    dateTextHint =
+      MutableLiveData<String>().apply {
+        val dateFormat = DateFormat.getDateFormat(requireContext()) as SimpleDateFormat
+        val pattern = dateFormat.toPattern()
+        val hint =
+          pattern
+            .replace("M", "MM") // Single-digit month to two-digit
+            .replace("d", "DD") // Single-digit day to two-digit
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt:68]: Will this work for other languages? If not, should we remove this as well?

**Review by shobhitagarwal1612 — APPROVED** [review]

**Pull request merged**